### PR TITLE
Update changelog.txt

### PR DESCRIPTION
**Pull request by trwalke**
Update changelog for MSAL 4.34.0

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -1,3 +1,14 @@
+4.34.0
+==========
+### Bug Fixes:
+**MSAL now has WithCcsRoutingHint(...)** to enable developers to more easily provide the CCS Routing hint during authentication. See issue [2725](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2725)
```
Comment (pmaytak) [inline changelog.txt:4]:
```suggestion
**MSAL now has `WithCcsRoutingHint()`** to enable developers to more easily provide the CCS routing hint during authentication. See issue [2725](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2725)
```

File: changelog.txt [hunk 2]
```diff
@@ -1,3 +1,14 @@
...
+
+### New Features
+**MSAL can now extract elements contained in wwwAuthenticate headers** to enable customers to more easily handle Continuous Access Evaluation and Conditional Access authentication context scenarios. See issue [#2679](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2679)
```
Comment (pmaytak) [inline changelog.txt:7]:
```suggestion
**MSAL can now extract elements contained in WWW-Authenticate headers** to enable customers to more easily handle Continuous Access Evaluation and Conditional Access authentication context scenarios. See issue [#2679](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2679)
```

File: changelog.txt [hunk 3]
```diff
@@ -1,3 +1,14 @@
...
+**WAM support is now GA in MSAL.NET.** WithExperimentalFeatures() is no longer required to authenticate with WAM. See https://aka.ms/msal-net-wam for more details
```
Comment (pmaytak) [inline changelog.txt:8]:
```suggestion
**WAM support is now generally available in MSAL.NET.** `WithExperimentalFeatures()` is no longer required to authenticate with WAM. See https://aka.ms/msal-net-wam for more details
```

File: changelog.txt [hunk 4]
```diff
@@ -1,3 +1,14 @@
...
+**MSAL enables Cache eviction more easily by exposing SuggestedCacheExpiry** which helps determining the cache eviction time. See issue [#2486](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2486)
```
Comment (pmaytak) [inline changelog.txt:9]:
```suggestion
**MSAL enables easier cache eviction by exposing `SuggestedCacheExpiry`** which helps determining the cache eviction time. See issue [#2486](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2486)
```

**Review by pmaytak — APPROVED** [review]

File: changelog.txt (hunk 2 above)
Comment (jennyf19) [reply changelog.txt:7]: I would suggest something like:
**MSAL.NET provides helper methods to extract the authentication parameters from the WWW-Authenticate headers**. This allows for dynamic scenarios such as claim challenge, Conditional Access Evaluation and Conditional Access authentication context scenarios. See https://aka.ms/msal-net/wwwAuthenticate and issue [#2679](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2679) for details.

File: changelog.txt (hunk 4 above)
Comment (jennyf19) [reply changelog.txt]: "determine" and maybe mention this is just for client_credentials

File: changelog.txt [hunk 5]
```diff
@@ -1,3 +1,14 @@
...
+**MSAL now adds runtime information to logs** enabling easier diagnosing of authenticaiton issues on all platforms. See issue [2559](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2559)
```
Comment (jennyf19) [inline changelog.txt:10]:
```suggestion
**MSAL now adds runtime information to logs** enabling easier diagnosing of authenticaition issues on all platforms. See issue [2559](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2559)
```

**Review by jennyf19 — APPROVED** [review]

**Pull request merged**